R/metadata_to_coldata.R

```diff
@@ -0,0 +1,75 @@
+#' Add sample metadata to colData of SCE object
+#'
+#' @param sce SingleCellExperiment object to add sample metadata to colData.
+#'   Must contain the `batch_column` as a column in `colData`.
+#' @param join_columns A character value giving the names of the columns in colData
+#'  to use for joining with the `sample_metadata`. Default is `library_id`
+#'
+#' @return SingleCellExperiment object with the contents of `metadata(sce)$sample_metadata`
+#'   added as new columns to the `colData(sce)`.
+#' @import SingleCellExperiment
+#'
```

Comment:
Tests are failing for me in part because `DataFrame` isn't in the environment, so add:
```suggestion
#' @importFrom S4Vectors DataFrame
```